fhirpath_helpers/elasticsearch/pytypes.py

```diff
@@ -8,297 +8,309 @@
+    Count = Quantity
+    Distance = Quantity
 
-fhir_data_types_maps = {
-    "boolean": Boolean,
-    "base64Binary": Token,
-    "integer": Integer,
-    "string": Text,
-    "decimal": Float,
-    "uri": Token,
-    "url": Token,
-    "canonical": Token,
-    "instant": Date,
-    "date": Date,
-    "dateTime": Date,
-    "time": Time,
-    "code": Token,
-    "oid": Token,
-    "id": Token,
-    "unsignedInt": Integer,
-    "positiveInt": Integer,
-    "uuid": Token,
-    "Attachment": Attachment,
-    "Coding": Coding,
-    "CodeableConcept": CodeableConcept,
-    "Quantity": Quantity,
-    "Distance": Distance,
-    "Count": Count,
-    "Money": Money,
-    "Duration": Duration,
-    "Range": Range,
-    "Ratio": Ratio,
-    "Period": Period,
-    "Identifier": Identifier,
-    "HumanName": HumanName,
-    "Address": Address,
-    "ContactPoint": ContactPoint,
-    "Timing": Timing,
-    "Dosage": Dosage,
-    "Meta": Meta,
-    "Annotation": Annotation,
-    "ContactDetail": ContactDetail,
-    "Age": Age,
-    "Reference": Reference,
-    "RelatedArtifact": RelatedArtifact,
-    "Signature": Signature,
-    "Population": Population,
-    "Narrative": Narrative,
-    "Expression": Expression,
-}
-fhir_data_types_maps_STU3 = deepcopy(fhir_data_types_maps)
-fhir_data_types_maps_STU3.update(
-    {
-        "Dosage": Dosage_STU3,
-        "Money": Money_STU3,
-        "RelatedArtifact": RelatedArtifact_STU3,
-        "Signature": Signature_STU3,
+    return {
+        "boolean": Boolean,
+        "base64Binary": Token,
+        "integer": Integer,
+        "string": Text,
+        "decimal": Float,
+        "uri": Token,
+        "url": Token,
+        "canonical": Token,
+        "instant": Date,
+        "date": Date,
+        "dateTime": Date,
+        "time": Time,
+        "code": Token,
+        "oid": Token,
+        "id": Token,
+        "unsignedInt": Integer,
+        "positiveInt": Integer,
+        "uuid": Token,
+        "Attachment": Attachment,
+        "Coding": Coding,
+        "CodeableConcept": CodeableConcept,
+        "Quantity": Quantity,
+        "Distance": Distance,
+        "Count": Count,
+        "Money": {FHIR_VERSION.STU3.name: Money_STU3, FHIR_VERSION.R4.name: Money}[
+            fhir_release
+        ],
+        "Duration": Duration,
+        "Range": Range,
+        "Ratio": Ratio,
+        "Period": Period,
+        "Identifier": Identifier,
+        "HumanName": HumanName,
+        "Address": Address,
+        "ContactPoint": ContactPoint,
+        "Timing": Timing,
+        "Dosage": {FHIR_VERSION.STU3.name: Dosage_STU3, FHIR_VERSION.R4.name: Dosage}[
+            fhir_release
+        ],
+        "Meta": Meta,
+        "Annotation": Annotation,
+        "ContactDetail": ContactDetail,
+        "Age": Age,
+        "Reference": Reference,
+        "RelatedArtifact": {
+            FHIR_VERSION.STU3.name: RelatedArtifact_STU3,
+            FHIR_VERSION.R4.name: RelatedArtifact,
+        }[fhir_release],
```

Comment:
I think an if / else would be clearer but I'm not sure if it would still be clearer with more than 2 cases